injector_test.py

```diff
@@ -1700,6 +1700,67 @@ def function(a: int) -> 'InvalidForwardReference':
     assert get_bindings(function) == {'a': int}
 
 
+def test_gets_bindings_for_annotated_type_with_inject_decorator() -> None:
+    UserID = Annotated[int, 'user_id']
+
+    @inject
+    def function(a: UserID, b: str) -> None:
+        pass
+
+    assert get_bindings(function) == {'a': UserID, 'b': str}
+
+
+def test_gets_bindings_of_annotated_type_with_inject_annotation() -> None:
+    UserID = Annotated[int, 'user_id']
+
+    def function(a: Inject[UserID], b: Inject[str]) -> None:
+        pass
+
+    assert get_bindings(function) == {'a': UserID, 'b': str}
+
+
+def test_gets_bindings_of_new_type_with_inject_annotation() -> None:
+    Name = NewType('Name', str)
+
+    @inject
+    def function(a: Name, b: str) -> None:
+        pass
+
+    assert get_bindings(function) == {'a': Name, 'b': str}
+
+
+def test_gets_bindings_of_inject_annotation_with_new_type() -> None:
+    def function(a: Inject[Name], b: str) -> None:
+        pass
+
+    assert get_bindings(function) == {'a': Name}
+
+
+def test_get_bindings_of_nested_noinject_inject_annotation() -> None:
+    # This is not how this is intended to be used
+    def function(a: Inject[NoInject[int]], b: NoInject[Inject[str]]) -> None:
```

Comment:
This may be a wild question, but are these usecases of `Inject[NoInject[...]` really considered practical use case?